src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -1214,7 +1216,7 @@ void InstanceKlass::initialize_impl(TRAPS) {
       call_class_initializer(THREAD);
     } else {
       // The elapsed time is so small it's not worth counting.
-      if (UsePerfData) {
+      if (UsePerfData || ProfileClassLinkage) {
```

Comment:
You have to have UsePerfData being true for this work so you don't need the change.

Reply:
Right, the `_perf_classes_inited` counter is a pre-existing counter whose creation depends on `UsePerfData`.